tests/unit/test_credentials.py

```diff
@@ -3165,6 +3166,45 @@ def test_can_pass_auth_token_from_file(self):
         self.assertEqual(creds.token, 'token')
         self.assertEqual(creds.method, 'container-role')
 
+    def test_reloads_auth_token_from_file(self):
+        token_file_path = os.path.join(self.tempdir, 'token.jwt')
+        with open(token_file_path, 'w') as token_file:
+            token_file.write('First auth token')
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE': token_file_path,
+        }
+        fetcher = self.create_fetcher()
+        timeobj = datetime.now(tzlocal())
+        fetcher.retrieve_full_uri.side_effect = [
+            {
+                "AccessKeyId": "first_key",
+                "SecretAccessKey": "first_secret",
+                "Token": "first_token",
+                "Expiration": (timeobj + timedelta(seconds=2)).isoformat(),
```

Comment:
nit; this could probably we 1s with a 1.5-2s sleep but that may end up creating more noise in the tests with false failures. Something to keep an eye on after release.